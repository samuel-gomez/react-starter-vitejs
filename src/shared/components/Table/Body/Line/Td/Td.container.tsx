import { ReactNode } from 'react';
import HelpHover, { THelpInfo } from 'shared/components/HelpInfo';
import Td from './Td';

export type TTdContainer = Omit<THelpInfo, 'children'> & {
  children?: ReactNode;
  label?: ReactNode;
  classModifier?: string;
  hover?: ReactNode;
  TdCmpt?: typeof Td;
  HelpHoverCmpt?: typeof HelpHover;
};

const TdContainer = ({ children, label, hover, classModifier, TdCmpt = Td, HelpHoverCmpt = HelpHover }: TTdContainer) => (
  <TdCmpt classModifier={classModifier}>
    <HelpHoverCmpt content={hover} classModifier="content">
      {label}
      {children}
    </HelpHoverCmpt>
  </TdCmpt>
);

export default TdContainer;
