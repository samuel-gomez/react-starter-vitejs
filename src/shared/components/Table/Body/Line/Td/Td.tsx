import { HelpInfo, Table as TableTk } from '@axa-fr/react-toolkit-all';
import type { ComponentPropsWithoutRef, ReactNode } from 'react';
import withClassNameModifier, { type TwithClassNameModifier } from 'shared/hoc/WithClassNameModifier';

export type TTdContainer = Omit<ComponentPropsWithoutRef<typeof HelpInfo>, 'children' | 'content'> & {
  children?: ReactNode;
  label?: ReactNode;
  hover?: ReactNode;
  TdCmpt?: typeof TableTk.Td;
  HelpHoverCmpt?: typeof HelpInfo;
} & TwithClassNameModifier;

const TdContainer = withClassNameModifier(
  ({ children, label, hover, className, HelpHoverCmpt = HelpInfo, ...rest }: TTdContainer) => (
    <td {...rest} className={className}>
      <HelpHoverCmpt content={hover} classModifier="content">
        {label}
        {children}
      </HelpHoverCmpt>
    </td>
  ),
  { defaultClassName: 'af-table__cell' },
);

export default TdContainer;
