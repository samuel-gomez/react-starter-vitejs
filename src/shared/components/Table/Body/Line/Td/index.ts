export { default } from './Td';
export type { TTdContainer } from './Td';
