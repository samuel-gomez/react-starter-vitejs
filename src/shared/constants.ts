import packageJson from '../../package.json';

/** ***********************************************************************
 * FORM CONSTANTS
 ************************************************************************ */
export const MSG_REQUIRED = 'Le champ est obligatoire';

export const MSG_LENGTH_3 = 'Le champ doit faire au moins 3 caractères';
export const MSG_FORMAT = 'Format invalide';
export const MSG_ERROR_FORMAT = 'Format de fichier incorrect';

export const DEFAULT_OPTION_LABEL = '- Sélectionner -';
export const DEFAULT_OPTION = {
  value: '',
  label: DEFAULT_OPTION_LABEL,
};

// patterns
export const patternEmail = /^[\w-+]+@([\w-+]+\.)+[\w-]{2,}$/;
export const patternPhoneNumber = /^[+0][\d ]*$/;
export const patternZipCode = /^\d{5}$/;
export const patternSiret = /^\d{14}$/;
export const patternSiren = /^\d{9}$/;
export const patternSiretSiren = /^\d{9,14}$/;

export const phoneNumberRule = {
  required: true,
  minLength: 4,
  maxLength: 20,
  pattern: patternPhoneNumber,
};

export const emailRule = {
  required: true,
  minLength: 6,
  maxLength: 60,
  pattern: patternEmail,
};

/** ***********************************************************************
 * API STATUS CONSTANTS
 ************************************************************************ */
export const STATUS_HTTP = {
  SUCCESS: 200,
  CREATED: 201,
  BAD_REQUEST: 400,
  UNAUTHORIZE: 401,
  FORBIDDEN: 403,
  NOTFOUND: 404,
  SERVER_ERROR: 500,
};

export const STATUS_API = {
  SUCCESS: STATUS_HTTP.SUCCESS,
  CREATED: STATUS_HTTP.CREATED,
  WARNING: 40,
  ERROR: 50,
};

/** ***********************************************************************
 * API URL
 ************************************************************************ */
export const API_URL = {
  BASE: 'base',
  GITHUB: 'github',
};

export const STATUS_HTTP_MESSAGES = {
  [STATUS_HTTP.SUCCESS]: 'Succès: La requête a réussi',
  [STATUS_HTTP.CREATED]: 'Succès: Création réussie',
  [STATUS_HTTP.BAD_REQUEST]: 'Erreur: La syntaxe de la requête est erronée',
  [STATUS_HTTP.UNAUTHORIZE]: 'Erreur: Une authentification est nécessaire pour accéder à la ressource',
  [STATUS_HTTP.FORBIDDEN]: 'Erreur: Accès non autorisé',
  [STATUS_HTTP.NOTFOUND]: 'Erreur: Elément non trouvé',
  [STATUS_HTTP.SERVER_ERROR]: 'Erreur: Problème technique ! Contacter votre support',
};

/** ***********************************************************************
 * USER CONSTANTS
 ************************************************************************ */

export const ADMIN = 'Admin';
export const READER = 'Reader';
export const WRITER = 'Writer';
export const USER = 'User';
export const ALL = '';

export const PROFILS = [ADMIN, READER, WRITER, USER, ALL];

/** ***********************************************************************
 * COMMONS CONSTANTS
 ************************************************************************ */

export const MODIFIER_CLASS = {
  success: 'success',
  disabled: 'disabled',
};

/** ***********************************************************************
 * VERSION CONSTANT
 ************************************************************************ */

const VERSION_TOOLKIT = 'master';

/** ***********************************************************************
 * DEMO CONSTANTS
 ************************************************************************ */

export const STORYBOOK = 'https://axafrance.github.io/react-toolkit/latest/storybook/?path=/story/';
export const DESIGN_SYSTEM = 'https://axafrance.github.io/design-system/';
export const GITHUB = `https://github.com/AxaGuilDEv/react-toolkit/tree/v${packageJson.dependencies['@axa-fr/react-toolkit-all']}/packages/`;
export const GITHUB_API = `AxaGuilDEv/react-toolkit/${VERSION_TOOLKIT}/packages/`;
