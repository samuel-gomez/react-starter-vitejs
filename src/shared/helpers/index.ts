export { formatDate, isValidDate, setDate } from './formatDate';
export { default as isEmptyOrNull, isEmptyArray, isEmptyObject } from './isEmptyOrNull';
export { default as mergeObj } from './mergeObj';
export { default as setAnomalyEmptyItems } from './setAnomalyEmptyItems';
export { default as setConfirmClassModifier } from './setConfirmClassModifier';
export { default as emptyFunction } from './emptyFunction';
export { default as setRequiredFieldRule, setRequiredMessage } from './setRequiredFieldRule';
