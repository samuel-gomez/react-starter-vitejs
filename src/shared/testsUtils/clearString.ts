const cleanString = (str?: string) => str?.replace(/\s/g, '').replace(/\n/g, '').replace(/\t/g, '').trim();

export default cleanString;
