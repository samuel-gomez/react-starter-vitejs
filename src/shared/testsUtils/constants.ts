const MOCK_API_URL = { base: 'https://your-url-api/' };

export default MOCK_API_URL;
