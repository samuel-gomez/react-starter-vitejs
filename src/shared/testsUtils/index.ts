export * from './customRender';
export { default as emptyFunction } from '../helpers/emptyFunction';
export { default as MOCK_API_URL } from './constants';
export { default as renderWithContainer } from './renderWithContainer';
