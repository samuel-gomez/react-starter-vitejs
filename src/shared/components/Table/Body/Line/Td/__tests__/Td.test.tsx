import { renderWithContainer, within } from 'shared/testsUtils';
import Td from '../Td';

const defaultProps = {
  children: null,
  label: 'label',
  hover: null,
};
const trContainer = document.createElement('tr');

describe('Td', () => {
  it('Render <Td/> without hover', () => {
    const { container } = renderWithContainer(<Td {...defaultProps}>child th</Td>, trContainer);
    within(container).getByText(/label/);
  });

  it('Render <Td/> with hover', () => {
    const { container } = renderWithContainer(
      <Td {...defaultProps} classModifier="custom" hover={<p>Hover content</p>}>
        child td
      </Td>,
      trContainer,
    );
    expect(container.getElementsByClassName('af-popover__wrapper').length).toBe(1);
  });
});
