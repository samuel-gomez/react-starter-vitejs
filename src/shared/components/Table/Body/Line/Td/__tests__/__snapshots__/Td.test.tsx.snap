// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Td > Render <Td/> with hover 1`] = `
<DocumentFragment>
  <div
    class="af-popover__wrapper"
  >
    <div
      class="af-popover__container af-popover__container--content"
    >
      <div
        class="af-popover__container-over"
        role="presentation"
      >
        labelchild td
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Td > Render <Td/> without hover 1`] = `
<DocumentFragment>
  labelchild th
</DocumentFragment>
`;
